rust/connlib/shared/src/messages.rs

```diff
@@ -217,13 +217,38 @@ pub struct ResourceDescriptionCidr {
     pub name: String,
 }
 
-#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq)]
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq, Hash)]
 #[serde(tag = "protocol", rename_all = "snake_case")]
 pub enum DnsServer {
     IpPort(IpDnsServer),
```

Comment:
Why the enum with one variant? To make serde serialize it differently?

Reply:
yep, just de-serialization. In the future we will probably have more variants though.